src/host/posix/dnssd.cpp

```diff
@@ -56,6 +56,11 @@ static otbr::DnssdPlatform::RegisterCallback MakeRegisterCallback(otInstance
     };
 }
 
+template <typename T> otbr::DnssdPlatform::DiscoveryCallbackIdentifier MakeDiscoveryCallbackIdentifier(T *aCallback)
```

Comment:
This function doesn't 'make' a new ID. Maybe we can rename it as `ToDiscoveryCallbackId()` for short?